### DISCUSSION TITLE
Best remesh strategy

### DISCUSSION THREAD

**Discussion by f-pucci** [pyvista/pyvista#4616] [Q&A]
Hello everyone,
I have generated a geometry through a scalar field and  the function "clip_scalar". 
Then, I have extruded one face of the geometry along an axis.

As I would need to use such geometry for CFD/FEA, I was wondering if you could suggest the best strategy to perform a remesh; for example, having all tetrahedral elements. 
Here are some examples of the current state of the mesh:

Extrusion:
![image](https://github.com/pyvista/pyvista/assets/126581990/923e5b1c-7a1a-4a8c-bdbb-e8e9ae6194cd)

Geometry:
![image](https://github.com/pyvista/pyvista/assets/126581990/ca36aa69-2d68-4f2f-af7d-f290a667a588)

As you can see, there are very obtuse triangles and the mesh quality is not good. 
Any advice is welcome!

Thank you in advance :)

**Comment by f-pucci** [comment]
Anyone that can help? :/

**Reply by banesullivan** [reply]
re-meshing is often a tough problem. Perhaps acvd might help: https://github.com/pyvista/pyacvd

**Reply by edabor** [reply]
Hi @f-pucci.

**[gmsh](https://gmsh.info/)** is a very powerful meshing sofwtare written in C++ with python wrappings (ie. as simple to install as `pip install gmsh`).

Here are all the python tutorials: https://gitlab.onelab.info/gmsh/gmsh/-/tree/master/tutorials/python

After a quick search, I found the following tutorials that relates to STL remeshing:
https://gitlab.onelab.info/gmsh/gmsh/-/blob/master/tutorials/python/t13.py
https://gitlab.onelab.info/gmsh/gmsh/-/blob/master/examples/api/remesh_stl.py

Hope that helps

**Reply by banesullivan** [reply]
Ah yes, I don't do re-meshing often so I forgot about gmsh
